src/client/Microsoft.Identity.Client/Cache/CacheImpl/InMemoryPartitionedCacheSerializer.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.;
+
+using System.Collections.Concurrent;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Core;
+
+namespace Microsoft.Identity.Client.Cache.CacheImpl
+{
+    /// <summary>
+    /// A simple partitioned cache, useful for Confidential Client flows. 
+    /// </summary>
+    internal class InMemoryPartitionedCacheSerializer
+        : AbstractPartitionedCacheSerializer
+    {
+        internal /* internal for test only */ ConcurrentDictionary<string, byte[]> CachePartition { get; }
+        private readonly ICoreLogger _logger;
+
+        public InMemoryPartitionedCacheSerializer(ICoreLogger logger)
+        {
+            CachePartition = new ConcurrentDictionary<string, byte[]>();
+            _logger = logger;
+        }
+
+        protected override byte[] ReadCacheBytes(string cacheKey)
+        {
+            if (CachePartition.TryGetValue(cacheKey, out byte[] blob))
+            {
+                return blob;
+            }
+
+            _logger.Verbose($"[InMemoryPartitionedTokenCache] ReadCacheBytes did not find cacheKey {cacheKey}");
+
+            return null;
+        }
+
+        protected override void RemoveKey(string cacheKey)
+        {
+            bool removed = CachePartition.TryRemove(cacheKey, out _);
```

Comment:
is it valuable to include the cache size in all the logs messages, if it's available?

Reply:
Added it on Write only, as `Count` is expensive on a ConcurrentDictionary